src/bin/sol-fbp-generator/main.c

```diff
@@ -233,8 +232,7 @@ read_meta_type_file(const struct sol_flow_metatype_context *ctx,
     }
 
     contents = sol_file_reader_get_all(reader);
-    *buf = contents.data;
-    *size = contents.len;
+    *buf = SOL_BUFFER_INIT_CONST((char *)contents.data, contents.len);
```

Comment:
the actual problem is the function signature, this `buf` is completely non-sense, likely it's a left over.

All that is using `buffer` could be modified to take a `sol_str_slice`, thus the `read_file` and `sol_flow_parse_buffer()`. Or even better, if we moved this to `struct sol_blob`, then one can even remove the shadow list `file_readers` by using `sol_file_reader_to_blob()`. Users should take a reference of the blob, use the `sol_str_slice_from_blob()` when using the parsing, then `sol_blob_unref()` when done, so the file can be closed.

The code is ancient and predates things like blobs, could use a review.  Can you do it @kaspersky ?